### PR TITLE
feat: watch register behalf

### LINKED ISSUE #40: feat: watch register behalf

### PR DESCRIPTION
**Pull request by chris13524**
# Description

- Add `watch_register_behalf()` function which allows passing-through `register_auth` payload from client
- Non-successful RPC requests will include the error body
- `just fmt` will auto-format instead of just `--check`
- `get_message_id()` refactor left-over from https://github.com/WalletConnect/WalletConnectRust/pull/39

Resolves #40

## How Has This Been Tested?

With archive refactor

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: relay_client/src/http.rs [hunk 1]
```diff
@@ -280,7 +288,11 @@ impl Client {
         let status = result.status();
 
         if !status.is_success() {
-            return Err(HttpClientError::InvalidHttpCode(status).into());
+            let body = match result.text().await {
```
Comment (heilhead) [inline relay_client/src/http.rs:291]: I'd use `result.text().await.unwrap_or_else(|err| format!("... error calling result.text(): {err:?}"))`. Also the message is a bit strange. Maybe it should be something like 'error reading response body'?

File: justfile [hunk 2]
```diff
@@ -66,7 +66,7 @@ fmt:
 
   if command -v cargo-fmt >/dev/null; then
     echo '==> Running rustfmt'
-    cargo +nightly fmt --all -- --check
+    cargo +nightly fmt --all
```
Comment (heilhead) [inline justfile:69]: Why the change? This was used for validation, along with clippy to make everything is all right before pushing. Formatting is usually done on the fly by the editor.

**Review by heilhead — APPROVED** [review]
Looks good! Added some comments.

File: justfile (hunk 2 above)
Comment (heilhead) [reply justfile:69]: I see. It hasn't been a problem for me, ever, in VSCode. Maybe you forgot to enable the `+nightly` feature in your VSCode addons?

Comment (chris13524) [reply justfile:69]: I can check this, but either way, wouldn't it be better to auto-format instead of just checking? What if my VS Code is out-of-sync with my CI/just tooling?

Comment (heilhead) [reply justfile:69]: Sure, as long as we still check formatting on the CI.

Comment (chris13524) [reply justfile:69]: Absolutely! No doubt about that :)

File: relay_client/src/http.rs (hunk 1 above)
Comment (chris13524) [reply relay_client/src/http.rs]: I don't know how I forgot about that function, fixed in [9729b6a](https://github.com/WalletConnect/WalletConnectRust/pull/41/commits/9729b6a209b304c75bf31cbc8ba71a5d51efcfba)

Three dots was suppose to clearly separate a body from a body error, but probably unnecessary.

Actually, I just pushed another commit that uses the Result value directly: [fec5ef6](https://github.com/WalletConnect/WalletConnectRust/pull/41/commits/fec5ef6dc67fa9fb37a3417d782839a0e13c14eb)
